File: src/main/java/mygame/editor/actions/CircleDrawer.java
package mygame.editor.actions;

import mygame.editor.ui.CustonPane;
import mygame.editor.model.AbstractModel;
import mygame.editor.model.CircleModel;
import mygame.editor.model.Point;

import java.util.List;

/**
 * Created by oleh on 3/18/17.
 */
public class CircleDrawer extends Action {


   CircleModel model ;
    public CircleDrawer(CustonPane parent, List<AbstractModel> model) {
        super(parent,model);
    }

    @Override
    public void mouseMoved(Point pos) {
        if(model != null) model.updateRadius(pos);

    }

    @Override
    public void mousePressed(Point pos) {
        if(model == null) {
            model = new CircleModel(pos);

            parent.addItem(model);
        }

    }

    @Override
    public void mouseReleased(Point pos) {
        if (model != null)models.add(model);
        model = null;
    }

    @Override
    public void finishDrawing() {

    }
}

File: src/main/java/mygame/editor/actions/shapes/CircleDrawer.kt
package mygame.editor.actions.shapes;

import mygame.editor.actions.Action
import mygame.editor.ui.CustomPane;
import mygame.editor.view.AbstractView;
import mygame.editor.view.CircleModel;
import mygame.editor.model.Point;


/**
 * Created by oleh on 3/18/17.
 */
class CircleDrawer (val pane:CustomPane, model:List<AbstractView>) : Action(pane,model){


   var model:CircleModel? = null


    override fun mouseMoved(pos:Point) {
        model?.updateRadius(pos)
    }


    override fun mousePressed(pos:Point ) {
        if(model == null) {
            model = CircleModel(pos);

            pane.addItem(model!!)
        }

    }

    override fun mouseReleased(pos: Point) {
        if (model != null)models.add(model);
        model = null
    }

    override fun finishDrawing() {

    }
}
